### PR TITLE
Enrollment note field

### PR DESCRIPTION
**Pull request by jnelsonjava**
Bridge-3095

- Adding tests for enrollment notes on create and update

### CODE REVIEW

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/EnrollmentTest.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
 import org.sagebionetworks.bridge.rest.api.StudiesApi;
 import org.sagebionetworks.bridge.rest.exceptions.EntityNotFoundException;
 import org.sagebionetworks.bridge.rest.model.Enrollment;
+import org.sagebionetworks.bridge.rest.model.EnrollmentDetail;
```
Comment (jnelsonjava) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/EnrollmentTest.java:29]: Not used

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/EnrollmentTest.java [hunk 2]
```diff
@@ -47,6 +48,7 @@ public void after() throws Exception {
         if (studyCoordinator != null) {
             studyCoordinator.signOutAndDeleteUser();
         }
+        // TODO: does the admin need to be signed out/deleted?
```
Comment (jnelsonjava) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/EnrollmentTest.java:51]: Do not delete the admin.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/EnrollmentTest.java [hunk 3]
```diff
@@ -84,21 +86,28 @@ public void test() throws Exception {
             enrollment.setExternalId(externalId);
             enrollment.setUserId(user.getUserId());
             enrollment.setConsentRequired(true);
+            enrollment.setNote("initial test enrollment note should not set");
```
Comment (jnelsonjava) [inline src/test/java/org/sagebionetworks/bridge/sdk/integration/EnrollmentTest.java:89]: Should actually work.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/EnrollmentTest.java [hunk 4]
```diff
@@ -84,21 +86,28 @@ public void test() throws Exception {
...
+            studiesApi.updateEnrollment(STUDY_ID_1, user.getUserId(), enrollment).execute();
             
             // Now shows up in paged api
             EnrollmentDetailList list = studiesApi.getEnrollments(STUDY_ID_1, "enrolled", false, null, null).execute().body();
             assertTrue(list.getItems().stream().anyMatch(e -> e.getParticipant().getIdentifier().equals(user.getUserId())));
-            
+            assertTrue(list.getItems().stream().anyMatch(e -> e.getParticipant().getIdentifier().equals(user.getUserId()) && e.getNote().equals(enrollment.getNote())));
+
             list = studiesApi.getEnrollments(STUDY_ID_1, null, false, null, null).execute().body();
             assertTrue(list.getItems().stream().anyMatch(e -> e.getParticipant().getIdentifier().equals(user.getUserId())));
```
[the remark this answers is not in the document]
Comment (jnelsonjava) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/EnrollmentTest.java:113]: Not needed.

**Review by alxdarksage — CHANGES_REQUESTED** [review]

**Review by alxdarksage — APPROVED** [review]

**Pull request merged**